### DISCUSSION TITLE
Keeping interactive widgets in tooltips without keeping them open

### DISCUSSION THREAD

**Discussion by PPakalns** [emilk/egui#4793] [Q&A] [answered]
egui 0.28 introduced:
Support interactive widgets in tool-tips https://github.com/emilk/egui/pull/4596 by [@emilk](https://github.com/emilk)

This change allows to hover tool-tip and modify interactive widget inside tool-tip. (tool-tips can be hovered and they do not close if original item is not hovered anymore).

I am implementing functionality where hovering a button opens a tool-tip that shows the preview of the window that will be opened.  I want to disable the functionality, which keeps tool-tip open when interactive widgets are present inside tool-tip.

I can disable widget interactivity by calling ui.disable() inside tool-tip, but it changes the styling of the tool-tip (everything is grayed out)

Is there another mechanism to revert to old functionality without UI look (styling) being changed?

Thank you for your help!

**Comment by emilk** [comment]
No, not right now. The thought was this: if you put an interactive widget in a tooltip, you should allow the user to click it. Putting an interactive widget where the user can't click it sounds like bad UX to me.
But you can make a PR to add a switch for this behavior in `Style` somewhere if you feel it is important.

**Reply by PPakalns** [reply]
Thank you for the quick answer.  Will try to workaround around this by modifying styles or maybe making a PR.

**Comment by PPakalns** [comment, marked as answer]
Simple working workaround:

Instead of using:
```rs
response.on_hover_ui(|ui| {

});
```

You can do:
```rs
if response.hovered() {
    response.show_tooltip_ui(|ui| { ... });
}
```
